staging/src/k8s.io/dynamic-resource-allocation/structured/internal/allocatortesting/allocator_testing.go

```diff
@@ -0,0 +1,3544 @@
+						map[resourceapi.QualifiedName]resourceapi.DeviceAttribute{
+							"special": {
+								BoolValue: ptr.To(true),
+							},
+						},
+					).withDeviceCounterConsumption(
+						deviceCounterConsumption(counterSet1,
+							map[string]resource.Quantity{
+								"cpu1": resource.MustParse("1"),
+							},
+						),
+						deviceCounterConsumption(counterSet2,
+							map[string]resource.Quantity{
+								"mem": resource.MustParse("10Gi"),
+							},
+						),
+					),
+					device(device2, nil,
+						map[resourceapi.QualifiedName]resourceapi.DeviceAttribute{
+							"special": {
+								BoolValue: ptr.To(false),
+							},
+						},
+					).withDeviceCounterConsumption(
+						deviceCounterConsumption(counterSet1,
+							map[string]resource.Quantity{
+								"cpu2": resource.MustParse("1"),
+							},
+						),
+						deviceCounterConsumption(counterSet2,
+							map[string]resource.Quantity{
+								"mem": resource.MustParse("10Gi"),
+							},
+						),
+					),
+					device(device3, nil,
+						map[resourceapi.QualifiedName]resourceapi.DeviceAttribute{
+							"special": {
+								BoolValue: ptr.To(false),
+							},
+						},
+					).withDeviceCounterConsumption(
+						deviceCounterConsumption(counterSet1,
+							map[string]resource.Quantity{
+								"cpu3": resource.MustParse("1"),
+							},
+						),
+						deviceCounterConsumption(counterSet2,
+							map[string]resource.Quantity{
+								"mem": resource.MustParse("10Gi"),
+							},
+						),
+					),
+				).withCounterSet(
+					counterSet(counterSet1,
+						map[string]resource.Quantity{
+							"cpu1": resource.MustParse("1"),
+							"cpu2": resource.MustParse("1"),
+							"cpu3": resource.MustParse("1"),
+						},
+					),
+					counterSet(counterSet2,
+						map[string]resource.Quantity{
+							"mem": resource.MustParse("30Gi"),
+						},
+					),
+				),
+			),
+			node: node(node1, region1),
+			expectResults: []any{allocationResult(
+				localNodeSelector(node1),
+				deviceAllocationResult(req0, driverA, pool1, device2, false),
+				deviceAllocationResult(req0, driverA, pool1, device3, false),
+				deviceAllocationResult(req1, driverA, pool1, device1, false),
+			)},
+		},
+	}
+
+	for name, tc := range testcases {
+		t.Run(name, func(t *testing.T) {
+			_, ctx := ktesting.NewTestContext(t)
+			g := gomega.NewWithT(t)
+
+			// Listing objects is deterministic and returns them in the same
+			// order as in the test case. That makes the allocation result
+			// also deterministic.
+			var classLister informerLister[resourceapi.DeviceClass]
+			for _, class := range tc.classes {
+				classLister.objs = append(classLister.objs, class.DeepCopy())
+			}
+			claimsToAllocate := slices.Clone(tc.claimsToAllocate)
+			allocatedDevices := slices.Clone(tc.allocatedDevices)
+			slices := slices.Clone(tc.slices)
+
+			allocator, err := newAllocator(ctx, tc.features, unwrap(claimsToAllocate...), sets.New(allocatedDevices...), classLister, slices, cel.NewCache(1))
```

Comment:
Can we compare the features supported by the allocator being tested and the features required by the test and just skip tests that are not supported by the allocator. That way, I think we should be able to mostly just add new tests to this file.

Reply:
Yes, that's the plan. I couldn't do it for your prioritized list improvement because it has not separate feature gate, but for consumable capacity that'll work. I can add the infrastructure now.